packages/run-protocol/src/runStake/runStakeKit.js

```diff
@@ -0,0 +1,351 @@
+// @ts-check
+import { Far } from '@endo/far';
+import { AmountMath, AssetKind } from '@agoric/ertp';
+import { assertProposalShape } from '@agoric/zoe/src/contractSupport/index.js';
+import { ceilMultiplyBy } from '@agoric/zoe/src/contractSupport/ratio.js';
+import { makeNotifierKit } from '@agoric/notifier';
+import { M, matches } from '@agoric/store';
+import { makeTracer } from '../makeTracer.js';
+import { applyDelta, assertOnlyKeys, transfer } from '../contractSupport.js';
+import { calculateCurrentDebt, reverseInterest } from '../interest-math.js';
+
+const { details: X, quote: q } = assert;
+
+const trace = makeTracer('R1');
+
+const VaultPhase = {
+  ACTIVE: /** @type {'Active'} */ ('Active'),
+  CLOSED: /** @type {'Closed'} */ ('Closed'),
+};
```

Comment:
not a vault. do we have a name yet for this thing that can be active or closed? it's in the state of `RUNStakeKit` but a kit shouldn't have a state. Maybe `StakingAccount`? 

Also `{const}` can save some typing
```suggestion
const AccountPhase = /** @type {const} */ {
  ACTIVE: 'Active',
  CLOSED: 'Closed',
};
```

Reply:
Very nearby we use "account" to refer to cosmos accounts. I'm trying out `LoanPhase`...